packages/core/content-releases/server/src/services/release.ts

```diff
@@ -481,14 +496,37 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
 
     // Only publish the release if all action updates are applied successfully to their entry, otherwise leave everything as is
     await strapi.db.transaction(async () => {
-      for (const contentTypeUid of Object.keys(actions)) {
+      // First we publish all the singleTypes
+      for (const { uid, action, id } of singleTypeActions) {
+        // @ts-expect-error - populateBuilderService should be a function but is returning service
+        const populate = await populateBuilderService(uid).populateDeep(Infinity).build();
+
+        const entry = await strapi.entityService.findOne(uid, id, { populate });
+
+        try {
+          if (action === 'publish') {
+            await entityManagerService.publish(entry, uid);
+          } else {
+            await entityManagerService.unpublish(entry, uid);
+          }
+        } catch (error) {
+          if (error instanceof errors.ApplicationError && error.message === 'already.published') {
```

Comment:
- what about already unpublished?
- why don't we have similar code for content types?

Reply:
Oh my bad, I didn't complete this part sorry!

collectionTypes are being published using `publishMany`, we don't return already.(un)published errors for `publishMany` or `unpublishMany`